From 56211631dfb3bb0c7eb85afa5e8077f3f7c4aaad Mon Sep 17 00:00:00 2001
From: Cheick Traore <cheick.traore@foss.st.com>
Date: Thu, 13 Feb 2025 10:52:36 +0100
Subject: [PATCH 1528/1695] drivers: remove alarm future from counter API

Alarm is no more used to trigger a callback. The framework is now
completely based on events.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 57eb8b9c5f93171359752bc82d35f32a3672a0c4 drivers: remove alarm future from counter API]
Change-Id: Ib2be4eda6dc2b54992df8bc652887065fe98d539
Signed-off-by: Cheick Traore <cheick.traore@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/404185
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 core/drivers/counter/counter.c | 38 ----------------------------------
 core/include/drivers/counter.h | 31 ---------------------------
 2 files changed, 69 deletions(-)

diff --git a/core/drivers/counter/counter.c b/core/drivers/counter/counter.c
index 4e8c9882a..a8ad40c0f 100644
--- a/core/drivers/counter/counter.c
+++ b/core/drivers/counter/counter.c
@@ -70,44 +70,6 @@ TEE_Result counter_get_value(struct counter_device *counter, unsigned int *ticks
 	return counter->ops->get_value(counter, ticks);
 }
 
-TEE_Result counter_set_alarm(struct counter_device *counter)
-{
-	struct alarm_cfg *alarm = NULL;
-	TEE_Result res = TEE_SUCCESS;
-
-	assert(counter && counter->ops);
-
-	alarm = &counter->alarm;
-
-	if (alarm->is_enabled || !alarm->callback || !counter->ops->set_alarm)
-		return TEE_ERROR_NOT_SUPPORTED;
-
-	if (alarm->ticks > counter->max_ticks)
-		return TEE_ERROR_NOT_SUPPORTED;
-
-	res = counter->ops->set_alarm(counter);
-	if (!res)
-		counter->alarm.is_enabled = true;
-
-	return res;
-}
-
-TEE_Result counter_cancel_alarm(struct counter_device *counter)
-{
-	TEE_Result res = TEE_SUCCESS;
-
-	assert(counter);
-
-	if (!counter->ops->cancel_alarm)
-		return TEE_ERROR_NOT_SUPPORTED;
-
-	res = counter->ops->cancel_alarm(counter);
-	if (!res)
-		counter->alarm.is_enabled = false;
-
-	return res;
-}
-
 TEE_Result counter_set_threshold(struct counter_device *counter,
 				 unsigned int ticks)
 {
diff --git a/core/include/drivers/counter.h b/core/include/drivers/counter.h
index 5c7556c5f..c5796736b 100644
--- a/core/include/drivers/counter.h
+++ b/core/include/drivers/counter.h
@@ -12,21 +12,6 @@
 #include <sys/queue.h>
 #include <tee_api_types.h>
 
-/**
- * @brief Alarm structure.
- *
- * @param callback Callback called on alarm (cannot be NULL).
- * @param ticks Number of ticks that triggers the alarm (absolute value).
- * @param priv Private data returned in callback.
- * @param is_enabled True if alarm is enabled, false otherwise.
- */
-struct alarm_cfg {
-	void (*callback)(unsigned int ticks, void *priv);
-	unsigned int ticks;
-	void *priv;
-	bool is_enabled;
-};
-
 enum counter_event_type {
 	/* Count value increased past ceiling */
 	COUNTER_EVENT_OVERFLOW,
@@ -68,8 +53,6 @@ struct counter_param {
  * @start: function to start the counter
  * @stop: function to stop the counter
  * @get_value: function to get the counter value
- * @set_alarm: (deprecated)
- * @cancel_alarm: (deprecated)
  * @set_threshold: function to set the counter threshold
  * @set_ceiling: function to set the counter ceiling
  * @enable_event:function to enable an event on counter value
@@ -81,8 +64,6 @@ struct counter_ops {
 	TEE_Result (*start)(struct counter_device *counter, void *config);
 	TEE_Result (*stop)(struct counter_device *counter);
 	TEE_Result (*get_value)(struct counter_device *counter, unsigned int *ticks);
-	TEE_Result (*set_alarm)(struct counter_device *counter);
-	TEE_Result (*cancel_alarm)(struct counter_device *counter);
 	TEE_Result (*set_threshold)(struct counter_device *counter,
 				    unsigned int ticks);
 	TEE_Result (*set_ceiling)(struct counter_device *counter,
@@ -105,7 +86,6 @@ struct counter_ops {
  * @param dev_list list on counter device.
  * @param is_used True if counter is used (exclusive consumer).
  * @param ops Operation table of the counter.
- * @param alarm Alarm configuration of the counter.
  * @param events List of event of the counter.
  * @param max_ticks Tick max value supported by the counter.
  * @param threshold Value reached on threshold event (COUNTER_EVENT_THRESHOLD).
@@ -118,7 +98,6 @@ struct counter_device {
 	LIST_ENTRY(counter_device) dev_list;
 	bool is_used;
 	const struct counter_ops *ops;
-	struct alarm_cfg alarm;
 	struct counter_event events[COUNTER_NB_EVENT];
 	unsigned int max_ticks;
 	unsigned int threshold;
@@ -141,16 +120,6 @@ TEE_Result counter_stop(struct counter_device *counter);
  */
 TEE_Result counter_get_value(struct counter_device *counter, unsigned int *ticks);
 
-/**
- * @brief Set an alarm.
- */
-TEE_Result counter_set_alarm(struct counter_device *counter);
-
-/**
- * @brief Cancel an alarm.
- */
-TEE_Result counter_cancel_alarm(struct counter_device *counter);
-
 /**
  * @brief Set counter threshold.
  */
-- 
2.47.2

